app/controllers/registrations_controller.rb

```diff
@@ -189,6 +189,9 @@ def do_update(require_password = true, confirm = false)
             successfully_updated = false
           else
             successfully_updated = current_user.update_with_password(password_update)
+            if !successfully_updated
+              message = _("Save unsuccessful. That email address is already registered. You must enter a unique email address.")
```

Comment:
Is it necissarily the case that a user for whom update_with_password fails has used a duplicate email address?
I could see this triggering on an incorrect password as well

Reply:
Hmm. We should look at refactoring this whole controller at some point. Its overly complex and doesn't need to be IMO. That's for another day though! ;)

This code executes when the user edits their profile. Specifically when they've changed their email address. If the password were invalid Devise kicks out to its own handler I believe. So the only other failure point would be a duplicate email address. The UI form itself should be preventing invalid email addresses.